scripts/check-legal-placeholders.ts

```
@@ -0,0 +1,192 @@
+    const lines = content.split("\n");
+
+    // Match patterns like [Something] or [Something Text]
+    const placeholderRegex = /\[([^\]]+)\]/g;
+
+    lines.forEach((line, index) => {
+      const matches = line.matchAll(placeholderRegex);
+      for (const match of matches) {
+         // Skip markdown links and legitimate brackets
+         const beforeMatch = line.substring(0, match.index);
+        // Skip markdown links and legitimate brackets
+        const beforeMatch = line.substring(0, match.index);
```

Biome (2.1.2)

[error] Shouldn't redeclare 'beforeMatch'. Consider to delete it or rename it.

'beforeMatch' is defined here:


(lint/suspicious/noRedeclare)